packages/jellyfish-transaction/src/script/defi/dftx_price.ts

```diff
@@ -0,0 +1,59 @@
+import BigNumber from 'bignumber.js'
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export interface CurrencyPair {
+  token: string // ---------------------| max 8 bytes
+  currency: string // -----------------| max 8 bytes
+}
+
+/**
+   * Composable CurrencyPair, C stands for Composable.
+   * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+   */
+export class CCurrencyPair extends ComposableBuffer<CurrencyPair> {
+  composers (cp: CurrencyPair): BufferComposer[] {
+    return [
+      ComposableBuffer.varUIntUtf8BE(() => cp.token, v => cp.token = v),
+      ComposableBuffer.varUIntUtf8BE(() => cp.currency, v => cp.currency = v)
+    ]
+  }
+}
+
+export interface TokenAmount {
+  currency: string // ---------------------| max 8 bytes
+  amount: BigNumber // -----------------| 8 bytes unsigned
+}
+
+/**
+   * Composable TokenAmount, C stands for Composable.
+   * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+   */
+export class CTokenAmount extends ComposableBuffer<TokenAmount> {
+  composers (tp: TokenAmount): BufferComposer[] {
+    return [
+      ComposableBuffer.varUIntUtf8BE(() => tp.currency, v => tp.currency = v),
+      ComposableBuffer.satoshiAsBigNumber(() => tp.amount, v => tp.amount = v)
```

Comment:
are you sure it's a satoshi unit? satoshiAsBigNumber is only used for 16 bytes, `0.00000000` calculation.